bz-manage.sh

```diff
@@ -381,4 +382,203 @@ get_bz_id_summary() {
   done
 }
 
+get_pr_info() {
+  local gh_pr gh_pr_info
+  gh_pr=$1
+  gh_pr_info=$(gh pr view "$gh_pr" --json title,body --jq .title,.body)
+  gh_pr_desc=$(echo "$gh_pr_info" | sed 1d | sed -e 's/[[:space:]]*$//' -e '/^$/d')
+  gh_pr_title=$(echo "$gh_pr_info" | sed -n 1p)
+  if [[ "$gh_pr_title" =~ ^feat.* ]]; then
+    gh_pr_type="Enhancement"
+  elif [[ "$gh_pr_title" =~ ^fix.* ]]; then
+    gh_pr_type="Bug Fix"
+  else
+    gh_pr_type="Enhancement"
+  fi
+}
+
+bz_add_doc_text() {
+  gh_pr=$1
+  bz=$2
+  get_pr_info "$gh_pr"
+  comment="Updating Doc Type and Text with info from the attached PR $gh_pr"
+  echo "BZ#$bz: $comment"
+  bugzilla modify --private \
+    --field cf_doc_type="$gh_pr_type" \
+    --field cf_release_notes="$gh_pr_desc" \
+    --comment "$comment" \
+    "$bz"
+}
+
+print_doc_text() {
+  doc_type=$1
+  doc_text=$2
+  echo "~~~"
+  echo "Doc Type:
+$doc_type"
+  echo -e "Doc Text:
+$doc_text"
+  echo "~~~"
+}
+
+bz_mod_doc_text() {
+  gh_pr=$1
+  bz=$2
+  query_file=$3
+  get_pr_info "$gh_pr"
+  doc_text="$(jq -r "select(.id == $bz) | .cf_release_notes" "$query_file" | sed -e 's/[[:space:]]*$//' -e '/^$/d')"
+  if [ -z "$doc_text" ]; then
+    comment="Updating Doc Type and Text with info from the attached PR $gh_pr"
+    echo "BZ#$bz: $comment"
+    bugzilla modify --private \
+      --field cf_doc_type="$gh_pr_type" \
+      --field cf_release_notes="$gh_pr_desc" \
+      --comment "$comment" \
+      "$bz"
+  else
+    if [ "$doc_text" != "$gh_pr_desc" ]; then
+      echo "BZ#$bz has Doc Text that differs from GitHub PR description"
+      echo "The current Doc Type and Text:"
+      print_doc_text "$doc_type" "$doc_text"
+      echo ""
+      echo "Doc Type and Text suggested from the GH PR:"
+      print_doc_text "$gh_pr_type" "$gh_pr_desc"
+      read -r -p 'Do you want to update the BZ with the suggested Doc Text and Type (y/n)? (default: n) ' update_bz </dev/tty
+      if [ "${update_bz:-n}" = n ]; then
+        return
+      elif [ "${update_bz:-n}" = y ]; then
+        comment="Updating Doc Type and Text with info from the attached PR $gh_pr"
+        echo "BZ#$bz: $comment"
+        bugzilla modify --private \
+          --field cf_doc_type="$gh_pr_type" \
+          --field cf_release_notes="$gh_pr_desc" \
+          --comment "$comment" \
+          "$bz"
+      fi
+    else
+      echo "BZ#$bz already has Doc Text that equals GitHub PR description"
+    fi
+  fi
+}
+
+process_count_prs() {
+  bz=$1
+  func=$2
+  args=$3
+  echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+  jq_count_prs="select(.id == $bz)\
+| [.external_bugs[]\
+| select((.type.type == \"GitHub\") and (.ext_bz_bug_id | contains(\"pull\")))]\
+| length"
+  gh_pr_count=$(jq -r "$jq_count_prs" "$query_file")
+  if [ "$gh_pr_count" -eq 1 ]; then
+    # shellcheck disable=SC2086
+    $func $args
```

Comment:
```suggestion
    "$func" "$@"
```

Reply:
I was doing it incorrectly, now using eval to re-evaluate the variables